tests/run.sh

```diff
@@ -10,37 +11,15 @@
 # 2. Run all needed commands for batch executing all python tests suites:
 #
 #       uv run tests/run_tests.py
+set -e
 
-find_binaries_and_run_tests() {
-
-    if [[ ! -v FLORESTA_TEMP_DIR ]]; then
-        # This helps us to keep track of the actual version being tested without conflicting with any already installed binaries.
-        HEAD_COMMIT_HASH=$(git rev-parse HEAD)
-
-        # Since its deterministic how we make the setup, we already know where to search for the binaries to be testing.
-        FLORESTA_TEMP_DIR="/tmp/floresta-functional-tests.${HEAD_COMMIT_HASH}"
-    fi
-
-    #prepare.sh will store the binaries in the same place, so we can find them.
-    FLORESTA_BIN_DIR="$FLORESTA_TEMP_DIR/binaries/florestad"
-    
-    UTREEXO_BIN_DIR="$FLORESTA_TEMP_DIR/binaries/utreexod"
-
-    # Here we save the original path from the bash session to restore it later. Cmon, we are not savages.
-    ORIGINAL_PATH=$PATH
-
-
-    # We add the generated binaries for testing to the PATH.
-    export PATH="$FLORESTA_BIN_DIR:$UTREEXO_BIN_DIR:$PATH"
-
-    $1
-
-    # Restores the original PATH
-    export PATH=$ORIGINAL_PATH
-
-}
-
-find_binaries_and_run_tests "uv run ./tests/run_tests.py"
+if [[ -z "$FLORESTA_TEMP_DIR" ]]; then
 
+    # This helps us to keep track of the actual version being tested without conflicting with any already installed binaries.
+    HEAD_COMMIT_HASH=$(git rev-parse HEAD)
 
+    # Since its deterministic how we make the setup, we already know where to search for the binaries to be testing.
+    export FLORESTA_TEMP_DIR="/tmp/floresta-functional-tests.${HEAD_COMMIT_HASH}"
 
+fi
+uv run ./tests/run_tests.py
```

Comment:
This is not passing the datadir, so it's writing to `/tmp/floresta-func-tests/` rather than `/tmp/floresta-functional-tests.${HEAD_COMMIT_HASH}`